nixos/modules/system/boot/networkd.nix

```diff
@@ -277,7 +278,7 @@ let
   checkDhcp = checkUnitConfig "DHCP" [
     (assertOnlyFields [
       "UseDNS" "RoutesToDNS" "UseNTP" "UseMTU" "Anonymize" "SendHostname" "UseHostname"
-      "Hostname" "UseDomains" "UseRoutes" "UseTimezone" "CriticalConnection"
+      "Hostname" "UseDomains" "UseRoutes" "UseTimezone"
```

Comment:
Upstream removed it from their docs, but they still allow setting it to keep compatibility.

We will now fail, and force users to migrate to the new setting - which might be the better decision, given we can have these checks at configuration time, and don't need to keep supporting old field indefinitely, as unknown fields are just ignored upstream.